R/sensivitity-time-profiles.R

```diff
@@ -87,92 +149,195 @@ sensitivityTimeProfiles <- function(sensitivityCalculation,
+    scaling = plotConfiguration$yAxisScale
+  )
+  cBreaks <- c(
+    min(data$ParameterFactor, na.rm = TRUE), 1,
+    max(data$ParameterFactor, na.rm = TRUE)
+  )
+  cBreaks <- unique(ifelse(cBreaks == 0, 1, cBreaks))
+
+  # map each parameter path to its own plot ----
+
+  plotList <- purrr::map(
+    unique(data$ParameterPath),
+    ~ {
+      dataSubset <- dplyr::filter(data, ParameterPath == .x)
+
+      # basic plot setup -------------------------
+
+      plot <- ggplot() +
+        geom_line(
+          data = dplyr::filter(dataSubset, ParameterFactor != 1.0),
+          aes(
+            x = Time,
+            y = Concentration,
+            group = ParameterFactor,
+            color = ParameterFactor
+          ),
+          linewidth = plotConfiguration$linesSize,
+          alpha = plotConfiguration$linesAlpha,
+          na.rm = TRUE
+        ) +
+        geom_line(
+          data = dplyr::filter(dataSubset, ParameterFactor == 1.0),
+          aes(Time, Concentration),
+          color = "black",
+          linewidth = plotConfiguration$linesSize,
+          alpha = plotConfiguration$linesAlpha,
+          na.rm = TRUE
+        )
+
+      # adjusting axis scales
+      if (isTRUE(plotConfiguration$xAxisScale == "log")) {
+        plot <- plot + scale_x_log10()
+      }
+      if (isTRUE(plotConfiguration$yAxisScale == "log")) {
+        plot <- plot +
+          scale_y_log10(
+            limits = replace(pLimits, pLimits == 0, 0.001),
+            breaks = pBreaks,
+            labels = scales::label_number()
+          )
+      } else {
+        plot <- plot +
+          scale_y_continuous(
+            limits = pLimits,
+            breaks = pBreaks,
+            labels = scales::label_number()
+          )
+      }
+
+      # finalize plot ----------------------------
+
+      # note: facet wrap on unique PK parameter to obtain facet titles
+      plot <- plot +
+        facet_wrap(~ParameterPath) +
+        labs(
+          x = plotConfiguration$xLabel,
+          y = plotConfiguration$yLabel,
+          title = NULL,
+          color = plotConfiguration$legendTitle
+        )
+
+      # theme adjustments
+      plot <- plot +
+        theme_bw(base_size = 11) +
+        theme(
+          legend.position = plotConfiguration$legendPosition,
+          panel.grid.minor = element_blank(),
+          text = element_text(size = 11)
+        ) +
+        guides(
+          color = guide_colorbar(
+            title = plotConfiguration$legendTitle,
+            ticks = FALSE,
+            draw.ulim = FALSE,
+            draw.llim = FALSE,
+            title.position = "top"
+          )
+        )
+
+      # apply color scales
+      if (is.null(plotConfiguration$linesColor)) {
+        plot <- plot +
+          colorspace::scale_color_continuous_diverging(
+            palette = "Berlin",
+            mid = log10(1),
+            transform = "log",
+            breaks = cBreaks
+          )
+      } else {
+        plot <- plot +
```

Comment:
I would maybe use a predefined diverging palette like Berlin available here: https://colorspace.r-forge.r-project.org/articles/hcl_palettes.html#diverging-palettes

Reply:
We have the same taste:

 ```
plot +
          colorspace::scale_color_continuous_diverging(
            palette = "Berlin",
            mid = log10(1),
            transform = "log",
            breaks = cBreaks
          )
```

Reply:
Oh, I missed that. Great !